### PR TITLE
[Hub Generated] Review request for Microsoft.Consumption to add version stable/2021-10-01

### PR DESCRIPTION
**Pull request by mitagarg**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/mitagarg/azure-rest-api-specs/dev-consumption-Microsoft.Consumption-2021-10-01-mita...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              "exchangeRateDate": "2021-10-01T00:00:00Z",
+              "invoiceId": "",
+              "previousInvoiceId": "",
+              "pricingCurrencyCode": "USD",
+              "product": "Virtual Machines D Series - D1 - US East",
+              "productIdentifier": "DZH318Z0BQ4B00FV",
+              "productOrderId": "a3db7880-70eb-4b4c-6a79-1425a058df5a",
+              "productOrderName": "Azure plan",
+              "publisherName": "Microsoft",
+              "publisherType": "1PP",
```
Comment (adwise3) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json:55]: @mitagarg This should not be 1pp. Should be either Microsoft or AWS.

@shrutis06 can you confirm?

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByDepartment.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              "dev": "tools"
+            },
+            "properties": {
+              "billingAccountId": "xxxxxxxx",
+              "billingAccountName": "Account Name 1",
+              "billingPeriodStartDate": "2019-03-01T00:00:00.0000000Z",
+              "billingPeriodEndDate": "2019-03-31T00:00:00.0000000Z",
+              "billingProfileId": "xxxxxxxx",
+              "billingProfileName": "Account Name 1",
+              "accountName": "Account Name 1",
```
Comment (adwise3) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByDepartment.json:27]: Where is BenefitId/BenefitName ?

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+              "env": "newcrp",
+              "dev": "tools"
+            },
+            "properties": {
+              "billingAccountId": "1234:56789",
+              "billingAccountName": "Account Name 1",
+              "billingPeriodStartDate": "2021-10-01T00:00:00.0000000Z",
+              "billingPeriodEndDate": "2019-10-31T00:00:00.0000000Z",
+              "billingProfileId": "2468",
+              "billingProfileName": "Account Name 1",
```
Comment (adwise3) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json:25]: Where is benefitId/benefitName?

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+              "chargeType": "Usage",
+              "additionalInfo": "{  \"UsageType\": \"ComputeHR\",  \"ImageType\": \"Windows Client BYOL\",  \"ServiceType\": \"Standard_D1\",  \"VMName\": null,  \"VMProperties\": null,  \"VCPUs\": 1,  \"CPUs\": 0}",
+              "costInBillingCurrency": 1.84763819095477,
+              "costInPricingCurrency": 1.84763819095477,
+              "exchangeRate": "1",
+              "exchangeRateDate": "2021-10-01T00:00:00Z",
+              "invoiceId": "",
+              "previousInvoiceId": "",
+              "pricingCurrencyCode": "USD",
+              "product": "Virtual Machines D Series - D1 - US East",
```
Comment (adwise3) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json:50]: Where is provider??

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json (hunk 1 above)
Comment (mitagarg) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json:55]: @adwise3 updated

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json (hunk 3 above)
Comment (mitagarg) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/examples/UsageDetailsListByMCABillingAccount.json:25]: @adwise3 udated

**Review by handeeeadiguzel — APPROVED** [review]
All IXP APIs look good!

**Comment by qwordy** [comment]
Required statuses must pass before merging. Swagger SDK Breaking Change Tracking failed.

**Review by basma-msft — APPROVED** [review]

**Comment by mitagarg** [comment]
> Required statuses must pass before merging. Swagger SDK Breaking Change Tracking failed.

hello, we had meeting with Jeff, and Jeff has approved the "Breaking change" above.

**Comment by msyyc** [comment]
approve for python breaking change.

**Comment by raosuhas** [comment]
>           "readOnly": true

Consider using extendible enum

---
Refers to: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/consumption.json:2134 in baae8f0. [](commit_id = baae8f0128823527ad49afae452a1d3c7930dff4, deletion_comment = False)

**Comment by raosuhas** [comment]
>         },

Same here. Consider using the same objet as well

---
Refers to: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-10-01/consumption.json:2176 in baae8f0. [](commit_id = baae8f0128823527ad49afae452a1d3c7930dff4, deletion_comment = False)

**Review by raosuhas — APPROVED** [review]
:shipit:

**Review by qwordy — APPROVED** [review]

**Pull request merged**